gst/tensor_ros_src/tensor_ros_src.cc

```diff
@@ -388,6 +398,80 @@ gst_tensor_ros_src_dispose (GObject * object)
   G_OBJECT_CLASS (parent_class)->dispose (object);
 }
 
+/**
+ * @brief Start handler
+ */
+static gboolean
+gst_tensor_ros_src_start (GstBaseSrc * src)
+{
+  GstTensorRosSrc *rossrc = GST_TENSOR_ROS_SRC (src);
+
+  silent_debug (rossrc, "start RosSubscriber");
+  rossrc->ros_sub = new TensorRosSub ("TensorRosSub", rossrc->topic_name,
+    rossrc, rossrc->freq_rate);
+  rossrc->ros_sub->Start (&rossrc->thread);
+
+  while (!rossrc->configured)
+    g_usleep (G_USEC_PER_SEC / rossrc->freq_rate);
+
+  silent_debug (rossrc, "tensor configuration is completed");
+  return TRUE;
+}
+
+/**
+ * @brief Stop handler
+ */
+static gboolean
+gst_tensor_ros_src_stop (GstBaseSrc * src)
+{
+  GstTensorRosSrc *rossrc = GST_TENSOR_ROS_SRC (src);
+
+  silent_debug (rossrc, "stop RosSubscriber");
+  rossrc->ros_sub->RequestStop ();
+  return TRUE;
+}
+
+/**
+ * @brief Get caps handler
+ */
+static GstCaps *
+gst_tensor_ros_src_get_caps (GstBaseSrc * src, GstCaps * filter)
+{
+  GstTensorRosSrc *rossrc = GST_TENSOR_ROS_SRC (src);
+  GstCaps *caps;
+  GstCaps *ret_caps;
+
+  caps = gst_tensor_caps_from_config (&rossrc->config);
+
+  if (caps) {
+    if (filter) {
+      ret_caps = gst_caps_intersect_full (filter, caps, GST_CAPS_INTERSECT_FIRST);
+    } else {
+      ret_caps = gst_caps_ref (caps);
+    }
+  }
+
+  silent_debug (rossrc, "caps: %" GST_PTR_FORMAT, ret_caps);
+  gst_caps_replace (&rossrc->caps, ret_caps);
+
+  gst_caps_unref (caps);
+  return ret_caps;
+}
+
+/**
+ * @brief Set caps handler
+ */
+static gboolean
+gst_tensor_ros_src_set_caps (GstBaseSrc * src, GstCaps * caps)
+{
+  GstTensorRosSrc *rossrc = GST_TENSOR_ROS_SRC (src);
+
+  gst_caps_replace (&rossrc->caps, caps);
```

Comment:
use `gst_pad_set_caps (src->srcpad, caps)` to set caps to the pad here??

Reply:
Right. I will.